### PR TITLE
refactors specs and added schema validation

### PR DESCRIPTION
**Pull request by andela-oakinniranye**

### CODE REVIEW

File: spec/support/api/schemas/addresses/address.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        "longitude": { "type": "float" },
+        "latitude": { "type": "float" },
+        "postal_code": { "type": "string" },
+        "active": { "type": "boolean" },
+        "formatted_address": { "type": "string" },
+        "company": {
+          "$ref": "../companies/company.json#/definitions/company"
+        }
+      },
+      "required": ["id", "address_1", "address_2", "city", "state", "country", "longitude", "latitude", "postal_code", "active", "formatted_address"]
```
Comment (oojewale) [inline spec/support/api/schemas/addresses/address.json]: Do we really need `address_2` to be required, just curious.

File: spec/support/json_validator.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+RSpec::Matchers.define :match_response_schema do |schema|
+  match do |response|
+    schema_directory = "#{Dir.pwd}/spec/support/api/schemas"
```
Comment (oojewale) [inline spec/support/json_validator.rb]: Not sure if the `...support/api/schemas` is necessary as the app is an API only app. I think this wouldn't be less meaningful: `...support/schemas` What do you think @andela-oakinniranye?

**Review by oojewale — COMMENTED** [review]
Nice. Am also thinking, are we considering authentication or not for the MVP? If yes, devise-token-auth, jwt...or what?

File: spec/support/api/schemas/addresses/address.json (hunk 1 above)
Comment (andela-oakinniranye) [reply spec/support/api/schemas/addresses/address.json]: not really, for the specs I believe it should be however, it shouldn't affect the structural organization of the data

File: spec/support/json_validator.rb (hunk 2 above)
Comment (andela-oakinniranye) [reply spec/support/json_validator.rb]: you're right! would work towards correcting that ASAP

**Comment by andela-oakinniranye** [comment]
For the MVP, authentication hasn't been considered yet! However, when we get to the bridge, I guess we'd cross it!

**Pull request merged**